src/main/java/com/artipie/asto/cache/CacheControl.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.asto.cache;
+
+import com.artipie.asto.Key;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Cache control.
+ * @since 0.24
+ */
+public interface CacheControl {
+
+    /**
+     * Standard cache controls.
+     * @since 0.24
+     */
+    enum Standard implements CacheControl {
+        /**
+         * Don't use cache, always invalidate.
+         */
+        NO_CACHE(item -> CompletableFuture.completedFuture(false)),
+        /**
+         * Always use cache, don't invalidate.
+         */
+        ALWAYS(item -> CompletableFuture.completedFuture(true));
+
+        /**
+         * Origin cache control.
+         */
+        private final CacheControl origin;
+
+        /**
+         * Ctor.
+         * @param origin Cache control
+         */
+        Standard(final CacheControl origin) {
+            this.origin = origin;
+        }
+
+        @Override
+        public CompletionStage<Boolean> validate(final Key item) {
+            return this.origin.validate(item);
+        }
+    }
+
+    /**
+     * Validate cached item: checks if cached value can be used or needs to be updated by fresh
+     * value.
+     * @param item Cached item
+     * @return True if cached item can be user, false if needs to be updated
```

Comment:
@g4s8 it seems there is a little typo, should be `used` instead of `user`

Reply:
@olegmoz thanks, fixed